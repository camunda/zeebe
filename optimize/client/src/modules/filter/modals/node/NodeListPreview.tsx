/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {ReactNode} from 'react';
import {Tag} from '@carbon/react';

import {Tooltip, ModdleElement} from 'components';
import {t} from 'translation';

interface NodeListPreviewProps {
  nodes: ModdleElement[];
  operator?: string;
  type: string;
}

export default function NodeListPreview({nodes, operator, type}: NodeListPreviewProps) {
  const previewList: JSX.Element[] = [];
  const createOperator = (name: ReactNode) => {
    return <span className="previewItemOperator"> {name} </span>;
  };

  nodes.forEach((selectedNode, idx) => {
    previewList.push(
      <li key={idx} className="previewItem">
        <span>
          {' '}
          {idx !== 0 &&
            createOperator(
              operator === 'not in'
                ? t('common.filter.list.operators.and')
                : t('common.filter.list.operators.or')
            )}
          <b>{selectedNode.name || selectedNode.id}</b>
        </span>
      </li>
    );
  });

  const parameterName = (
    <Tag type="blue" className="parameterName">
      {t(
        'common.filter.nodeModal.preview.' + (operator === 'not in' ? 'notExecutedFlowNodes' : type)
      )}
    </Tag>
  );

  return (
    <>
      <Tooltip content={parameterName} overflowOnly>
        {parameterName}
      </Tooltip>
      <ul className="previewList filterText">{previewList}</ul>
    </>
  );
}
