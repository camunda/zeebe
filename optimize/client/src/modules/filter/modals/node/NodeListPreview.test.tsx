/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {shallow} from 'enzyme';

import NodeListPreview from './NodeListPreview';

const props = {
  nodes: [{id: 'bar', name: 'foo'}],
  operator: 'in',
  type: 'executedFlowNodes',
};

it('should create preview of nodes', () => {
  const node = shallow(<NodeListPreview {...props} />);

  expect(node).toMatchSnapshot();
});

it('should show the id of the flow node if the name is null', () => {
  const node = shallow(<NodeListPreview {...props} nodes={[{id: 'bar', name: undefined}]} />);

  expect(node.find('b')).toMatchSnapshot();
});

it('should create preview of selected nodes linked by or', () => {
  const flowNode1 = {
    name: 'foo',
    id: 'bar',
  };

  const flowNode2 = {
    name: 'foo',
    id: 'bar',
  };
  const node = shallow(<NodeListPreview {...props} nodes={[flowNode1, flowNode2]} />);

  expect(node).toMatchSnapshot();
});

it('should create preview of selected nodes linked by nor', () => {
  const flowNode1 = {
    name: 'foo',
    id: 'bar',
  };

  const flowNode2 = {
    name: 'foo',
    id: 'bar',
  };

  const changedProps = {
    nodes: [flowNode1, flowNode2],
    operator: 'not in',
  };

  const node = shallow(<NodeListPreview {...props} {...changedProps} />);

  expect(node).toMatchSnapshot();
});

it('should show executing node filter', () => {
  const node = shallow(
    <NodeListPreview {...props} operator={undefined} type="executingFlowNodes" />
  );

  expect(node.find('.parameterName').dive().find('Text').dive()).toIncludeText('Running');
});
