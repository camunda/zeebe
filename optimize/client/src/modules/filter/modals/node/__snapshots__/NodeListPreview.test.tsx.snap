// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create preview of nodes 1`] = `
<Fragment>
  <Tooltip
    content={
      <Tag
        className="parameterName"
        type="blue"
      >
        Running, canceled, or completed flow nodes
      </Tag>
    }
    overflowOnly={true}
  >
    <Tag
      className="parameterName"
      type="blue"
    >
      Running, canceled, or completed flow nodes
    </Tag>
  </Tooltip>
  <ul
    className="previewList filterText"
  >
    <li
      className="previewItem"
      key="0"
    >
      <span>
         
        <b>
          foo
        </b>
      </span>
    </li>
  </ul>
</Fragment>
`;

exports[`should create preview of selected nodes linked by nor 1`] = `
<Fragment>
  <Tooltip
    content={
      <Tag
        className="parameterName"
        type="blue"
      >
        Not executed flow nodes
      </Tag>
    }
    overflowOnly={true}
  >
    <Tag
      className="parameterName"
      type="blue"
    >
      Not executed flow nodes
    </Tag>
  </Tooltip>
  <ul
    className="previewList filterText"
  >
    <li
      className="previewItem"
      key="0"
    >
      <span>
         
        <b>
          foo
        </b>
      </span>
    </li>
    <li
      className="previewItem"
      key="1"
    >
      <span>
         
        <span
          className="previewItemOperator"
        >
           
          and
           
        </span>
        <b>
          foo
        </b>
      </span>
    </li>
  </ul>
</Fragment>
`;

exports[`should create preview of selected nodes linked by or 1`] = `
<Fragment>
  <Tooltip
    content={
      <Tag
        className="parameterName"
        type="blue"
      >
        Running, canceled, or completed flow nodes
      </Tag>
    }
    overflowOnly={true}
  >
    <Tag
      className="parameterName"
      type="blue"
    >
      Running, canceled, or completed flow nodes
    </Tag>
  </Tooltip>
  <ul
    className="previewList filterText"
  >
    <li
      className="previewItem"
      key="0"
    >
      <span>
         
        <b>
          foo
        </b>
      </span>
    </li>
    <li
      className="previewItem"
      key="1"
    >
      <span>
         
        <span
          className="previewItemOperator"
        >
           
          or
           
        </span>
        <b>
          foo
        </b>
      </span>
    </li>
  </ul>
</Fragment>
`;

exports[`should show the id of the flow node if the name is null 1`] = `
<b>
  bar
</b>
`;
